repositories/droplet_repository.go

```diff
@@ -0,0 +1,90 @@
+package repositories
+
+import (
+	"context"
+	"errors"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	workloadsv1alpha1 "code.cloudfoundry.org/cf-k8s-controllers/apis/workloads/v1alpha1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type DropletRecord struct {
+	GUID            string
+	State           string
+	CreatedAt       string
+	UpdatedAt       string
+	DropletErrorMsg string
+	Lifecycle       Lifecycle
+	Stack           string
+	ProcessTypes    map[string]string
+	AppGUID         string
+	PackageGUID     string
+	Labels          map[string]string
+	Annotations     map[string]string
+}
+
+type DropletRepo struct{}
+
+func (r *DropletRepo) FetchDroplet(ctx context.Context, k8sClient client.Client, dropletGUID string) (DropletRecord, error) {
+	buildList := &workloadsv1alpha1.CFBuildList{}
+	err := k8sClient.List(ctx, buildList)
+	if err != nil { // untested
+		return DropletRecord{}, err
+	}
+	allBuilds := buildList.Items
+	matches := filterBuildsByMetadataName(allBuilds, dropletGUID)
+
+	return r.returnDroplet(matches)
+}
+
+func (r *DropletRepo) returnDroplet(builds []workloadsv1alpha1.CFBuild) (DropletRecord, error) {
+	if len(builds) == 0 {
+		return DropletRecord{}, NotFoundError{}
+	}
+	if len(builds) > 1 { //untested
+		return DropletRecord{}, errors.New("duplicate builds exist")
+	}
+
+	cfBuild := builds[0]
+	stagingStatus := getConditionValue(&cfBuild.Status.Conditions, StagingConditionType)
+	succeededStatus := getConditionValue(&cfBuild.Status.Conditions, SucceededConditionType)
+	if stagingStatus == metav1.ConditionFalse &&
+		succeededStatus == metav1.ConditionTrue {
+		return r.cfBuildToDropletRecord(cfBuild), nil
+	}
+	return DropletRecord{}, NotFoundError{}
+}
+
+func (r *DropletRepo) cfBuildToDropletRecord(cfBuild workloadsv1alpha1.CFBuild) DropletRecord {
```

Comment:
This doesn't use any of the information on the DropletRepo, so it should be a message without a receiver.

Reply:
Addressed on commit [8a5dfced](https://github.com/cloudfoundry/cf-k8s-api/pull/131/commits/8a5dfced2f8a16a0d25310f384e2cb265d2a4a7d)